api/src/home/analysis-platform/job_handlers/azure_container_instances/__init__.py

```
@@ -55,60 +55,75 @@ def setup_service(self, service_id: str) -> str:
+                + "RegistryUsername: 'None'"
+            )
+            command_list = [
+                "/code/init.sh",
+                f"--input-id={self.data_source}/{self.job.entity['applicationInput']['_id']}",
+            ]
+            if reference_target:
+                command_list.append(f"--reference-target={reference_target}")
+            compute_resources = ResourceRequests(memory_in_gb=1.5, cpu=1.0)
+            container = Container(
+                name=self.azure_valid_container_name,
+                image=full_image_name,
+                resources=ResourceRequirements(requests=compute_resources),
+                command=command_list,
+                environment_variables=env_vars,
+            )
+            image_registry_credential = None
+
+            # Configure the container group
+            group = ContainerGroup(
+                location="norwayeast",
+                containers=[container],
+                os_type=OperatingSystemTypes.linux,
+                restart_policy=ContainerGroupRestartPolicy.never,
+                image_registry_credentials=image_registry_credential,
+            )
 
-        # Create the container group
-        result = self.aci_client.container_groups.begin_create_or_update(
-            config.AZURE_JOB_RESOURCE_GROUP, self.azure_valid_container_name, group
-        )
+            # Create the container group
+            result = self.aci_client.container_groups.begin_create_or_update(
+                config.AZURE_JOB_RESOURCE_GROUP, self.azure_valid_container_name, group
+            )
+
+            return result.status()
+
+        except (KeyError, AttributeError) as error:
+            raise Exception(
+                f"Job entity used as input to azure container jobs is missing: {error}. Please make required changes and create a new job."
+            )
+        except Exception as error:
+            raise Exception(f"Error occurred when staring azure container job: {error}")
```

No need for a catch all exceptions here. That will, and should, be handled by the caller function.